addons/alerting.libsonnet

```diff
@@ -10,7 +10,159 @@ function(config) {
     "If 'alerting' is set, 'slackWebhookURLCritical' or 'pagerdutyRoutingKey' should be declared"
   ),
 
-  local criticalReceiver =
+  local IDECriticalReceiver =
+    if
+      (std.objectHas(config.alerting, 'IDE') && std.objectHas(config.alerting.IDE, 'pagerdutyRoutingKey')) ||
+      (!std.objectHas(config.alerting, 'IDE') && std.objectHas(config.alerting, 'pagerdutyRoutingKey'))
+    then
+      |||
+        pagerduty_configs:
+          - send_resolved: true
+            routing_key: '%(pagerdutyRoutingKey)s'
+      ||| % {
+        pagerdutyRoutingKey: if std.objectHas(config.alerting, 'IDE') && std.objectHas(config.alerting.IDE, 'pagerdutyRoutingKey') then config.alerting.IDE.pagerdutyRoutingKey
+        else config.alerting.pagerdutyRoutingKey,
+      }
+    else
+      |||
+        slack_configs:
```

Comment:
Would it be possible to extract this to a local variable too and then reuse it for all of the receivers? Something like

```jsonnet
local slackConfig = |||
  slack_configs:
    - send_resolved: true
      api_url: %(slackWebhookUrlCritical)s
      channel: '%(slackChannelPrefix)s_critical'
      title: '[{{ .Status | toUpper }}{{ if eq .Status "firing" }}{{ end }}] %(clusterName)s Monitoring'
      text: |
        {{ range .Alerts }}
        **Please take immediate action!**
        *Cluster:* {{ .Labels.cluster }}
        *Alert:* {{ .Labels.alertname }}
        *Description:* {{ .Annotations.description }}
        {{ end }}
      actions:
      - type: button
        text: 'Runbook :book:'
        url: '{{ .CommonAnnotations.runbook_url }}'
|||

local IDECriticalReceiver = if .. 
  then ... 
  else slackConfig % {
    // all params goes here
  }
```

Reply:
Probably 🙂